barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -306,17 +358,16 @@ AvmTraceBuilder::AvmTraceBuilder(AvmPublicInputs public_inputs,
                                  std::vector<FF> calldata)
     // NOTE: we initialise the environment builder here as it requires public inputs
     : calldata(std::move(calldata))
-    , new_public_inputs(public_inputs)
+    , public_inputs(public_inputs)
```

Comment:
You could rename the passed public_inputs to public_inputs_, use std::move (like we do for execution_hints) and note that you should not change anything for merkle_tree_trace_builder()